src/lds.cpp

```diff
@@ -146,15 +146,149 @@ bool CldsPlayer::load(const std::string &filename, const CFileProvider &fp)
+      else
+        comword = 0x8001;
+      comhi = comword >> 8;
+      comlo = comword & 0xff;
+
+      if(comword) {
+        if(comhi == 0x80) {
+          crow += comlo;
+        } else {
+          if(comhi >= 0x80) {
+            switch(comhi) {
+              case 0xff:
+                command = TrackedCmdOPLCarrierModulatorVolume;
+                param = comlo;
+                break;
+              case 0xfe:
+                command = TrackedCmdTempo;
+                param = comword & 0x3f;
+                break;
+              case 0xfd:
+                volume = comlo;
+                break;
+              case 0xfc:
+                command = TrackedCmdNoteCut;
+                break;
+              case 0xfb:
+                command = TrackedCmdRetrigger;
+                param = 1;
+                break;
+              case 0xfa:
+                command = TrackedCmdPatternBreak;
+                break;
+              case 0xf9:
+                command = TrackedCmdPatternJumpTo;
+                param = comlo & maxpos;
+                break;
+              case 0xf8:
+                // lastune = 0 ???
+                break;
+              case 0xf7:
+                command = TrackedCmdVibrato;
+                param = comlo;
+                break;
+              case 0xf6:
+                command = TrackedCmdTonePortamento;
+                note = comlo + 12;
+                param = 0;
+                break;
+              case 0xf5:
+                command = TrackedCmdPitchSlideUpDown; // not a perfect match for c->finetune = comlo
+                param = comlo;
+                break;
+              case 0xf4:
+                command = TrackedCmdGlobalVolume;
+                param = comlo;
+                break;
+              case 0xf3:
+                // fadeonoff = comlo ??
+                command = TrackedCmdVolumeFadeIn;
+                param = comlo;
+                break;
+              case 0xf2:
+                command = TrackedCmdOPLTremolo;
+                param = comlo;
+                break;
+              case 0xf1:	// panorama
+              case 0xf0:	// progch
+                break;
+              default:
+                if(comhi < 0xa0) {
+                  command = TrackedCmdTonePortamento;
+                  // this note might be off..
+                  note = (comhi & 0x1f) + 12;
+                  param = 0;
+                }
+                break;
+            }
+          } else {
+            //unsigned char  sound;
+            unsigned short high;
+            signed char transp = transpose & 127;
+
+            if(transpose & 64) transp |= 128;
+
+            if(transpose & 128) {
+              //sound = (comlo + transp) & maxsound;
+              high = comhi << 4;
+            } else {
+              //sound = comlo & maxsound;
+              high = (comhi + transp) << 4;
+            }
+
+            if(chandelay[chan]) {
```

Comment:
```
src\lds.cpp(271): error C4700: uninitialized local variable 'chan' used
```

Reply:
I guess I have two options here.

a) easy way - remove the entire if() and not fix chandelay

b) add chandelay effect if a note is played and there are no other effects. (The chandelay seem to be a global feature per channel on this fileformat)